src/main/java/ox/app/states/GameState.java

```diff
@@ -15,8 +14,4 @@
      * could return new turn, for opposite player.
      */
     GameState nextState(String input) throws WrongArgumentException;
```

Comment:
Are you sure that all of your states will throw WrongArgumentException and will need an input?

It does not look like that in code

Reply:
It is needed in interface to mark that this method could throw some exceptions. It is compile time error to not do so.

Reply:
So design should be changed